### PR TITLE
rtags 1.2 (new formula)

### PR DESCRIPTION
**Pull request by smoofra**
rtags is a ctags-like source code cross-referencer with a clang frontend.

from rtags README:

RTags is a client/server application that indexes C/C++ code and keeps a persistent in-memory database of references, declarations, definitions, symbolnames etc. There’s also limited support for ObjC/ObjC++. It allows you to find symbols by name (including class and namespace scope). Most importantly we give you proper follow-symbol and find-references support. We also have neat little things like rename-symbol, integration with clang’s “fixits” (http://clang.llvm.org/diagnostics.html). We also integrate with flymake using clang’s vastly superior errors and warnings. Since RTags constantly will reindex “dirty” files you get live updating of compiler errors and warnings. Since we already know how to compile your sources we have a way to quickly bring up the preprocessed output of the current source file in a buffer.

While existing taggers like gnu global, cscope, etags, ctags etc do a good job for C they often fall a little bit short for C++. With its incredible lexical complexity, parsing C++ is an incredibly hard task and we make no bones about the fact that the only reason we are able to improve on the current tools is because of clang (http://clang.llvm.org/). RTags is named RTags in recognition of Roberto Raggi on whose C++ parser we intended to base this project but he assured us clang was the way to go. The name stuck though.

### CODE REVIEW

File: Library/Formula/rtags.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
+
+class Rtags < Formula
+  homepage "https://github.com/Andersbakken/rtags"
+
+  stable do
+    url "https://github.com/Andersbakken/rtags/archive/v1.1.tar.gz"
+    sha1 "305f722b4fa1fc06954be31483d01c129083de1c"
+    patch :DATA
```
Comment (mikemcquaid) [inline Library/Formula/rtags.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

Comment (mikemcquaid) [inline Library/Formula/rtags.rb]: Also, please provide another newline after `patch` and before `resource`.

File: Library/Formula/rtags.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
+
```
Comment (mikemcquaid) [inline Library/Formula/rtags.rb]: Remove empty line.

File: Library/Formula/rtags.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    patch :DATA
+    resource "rct" do
+      url "https://github.com/Andersbakken/rct.git", :revision => "39ee9faf055256b3942297b18117841d90bb0ef5"
+    end
+  end
+
+  head "https://github.com/Andersbakken/rtags.git"
+
+  depends_on "cmake" => :build
+  depends_on "llvm" => "with-clang"
```
Comment (mikemcquaid) [inline Library/Formula/rtags.rb]: Can't this use the Clang provided with Xcode?

File: Library/Formula/rtags.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      (buildpath/"src/rct").install resource("rct")
+    end
+
+    ENV.prepend_path "PATH", "#{opt_libexec}/llvm/bin"
+
+    mkdir "build" do
+      args = std_cmake_args
+      args << ".."
+      if build.stable?
+        args.delete("-DCMAKE_BUILD_TYPE=None")
```
Comment (mikemcquaid) [inline Library/Formula/rtags.rb]: Why is this needed?

File: Library/Formula/rtags.rb (hunk 1 above)
Comment (smoofra) [inline Library/Formula/rtags.rb]: The patch is a back port from upstreams's "master" branch.   They've already got the fix.

File: Library/Formula/rtags.rb (hunk 3 above)
Comment (smoofra) [inline Library/Formula/rtags.rb]: nope, the cmake script will fail.

File: Library/Formula/rtags.rb (hunk 4 above)
Comment (smoofra) [inline Library/Formula/rtags.rb]: another cmake bug that's fixed in 'master'

File: Library/Formula/rtags.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Formula/rtags.rb]: Cool! Please add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Library/Formula/rtags.rb (hunk 3 above)
Comment (mikemcquaid) [inline Library/Formula/rtags.rb]: It would be good to try and use the system Clang if at all possible.

File: Library/Formula/rtags.rb (hunk 4 above)
Comment (mikemcquaid) [inline Library/Formula/rtags.rb]: Ok, needs a comment explaining that, thanks!

File: Library/Formula/rtags.rb (hunk 1 above)
Comment (smoofra) [inline Library/Formula/rtags.rb]: upstream just made a new 1.2 tag with the cmake fix, so i'll just bump it to that

File: Library/Formula/rtags.rb (hunk 3 above)
Comment (smoofra) [inline Library/Formula/rtags.rb]: upstream seems to recommend using brew's llvm.    I don't think it's possible.

Comment (smoofra) [inline Library/Formula/rtags.rb]: confirmed from upstream, using native llvm is not possible.

**Comment by DomT4** [comment]
If still interested here please:

* Add a short (80 characters or less) `desc` above `homepage`.
* Update to the latest version
* Fix any other nits highlighted by `brew audit --strict`.

Thanks!

File: Library/Formula/rtags.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  head "https://github.com/Andersbakken/rtags.git"
+
+  depends_on "cmake" => :build
+  depends_on "llvm" => "with-clang"
+  depends_on "openssl"
+
+  def install
+    unless build.head?
```
Comment (DomT4) [inline Library/Formula/rtags.rb]: This should be `if build.stable?` since there's no `devel`.

**Comment by DomT4** [comment]
Closing due to no response. Will welcome a new PR with the issues highlighted addressed. Thanks for your submission to Homebrew!
